### PR TITLE
[protobuf] Update to 3.7.1

### PR DESCRIPTION
**Pull request by myd7349**
Tested with:

* [x] x86-windows
* [x] x86-windows-static
* [x] x64-windows
* [x] x64-windows-static
* [x] x64-linux

Test project: https://github.com/myd7349/Ongoing-Study/tree/master/cpp/protobuf/person_example

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@myd7349, thanks for contributing to vcpkg!

Here is fail triplets(x86-windows and x64-windows ) and regression(3fd) with your PR.
```
 Processing x64-windows                                          849 vs 850
     3fd                           **regression**                Fail vs Pass
 Processing x86-windows                                          829 vs 830
     3fd                           **regression**                Fail vs Pass
```

**Comment by myd7349** [comment]
Hi! @PhoebeHui Thanks for your reply.

I am a bit confused about the CI result. Does it mean that after this update the installation of `google-cloud-cpp` and `orc` will fail?

I will check it out.

**Comment by aahmed-se** [comment]
any update on this ?

**Comment by myd7349** [comment]
I tried to install `google-cloud-cpp` on my PC and it failed:

>E:\vcpkg\buildtrees\google-cloud-cpp\x86-windows-dbg\external\nlohmann_json\include\google/cloud/storage/internal/nlohmann_json.hpp(3605): error C2039: '_strtoui64': is not a member of 'std'

**Comment by myd7349** [comment]
I have created a PR to fix this problem: https://github.com/protocolbuffers/protobuf/pull/5864

**Comment by PhoebeHui** [comment]
@myd7349, thanks for the updates! Check latest cl results, orc failed with x64-osx triplet:
```
 Processing arm-uwp                        412 vs 412
 Processing arm64-windows                  472 vs 472
 Processing x64-linux                      617 vs 617
 Processing x64-osx                        609 vs 610
     orc                           **regression**                Fail vs Pass
 Processing x64-uwp                        440 vs 440
 Processing x64-windows                    886 vs 886
 Processing x64-windows-static             780 vs 780
 Processing x86-windows                    866 vs 866
```

**Comment by myd7349** [comment]
OK. I will try to resolve it.

**Comment by myd7349** [comment]
`orc` build error on osx: [install-x64-osx-dbg-out.log](https://github.com/Microsoft/vcpkg/files/2973701/install-x64-osx-dbg-out.log)

Seems an known issue: https://github.com/protocolbuffers/protobuf/pull/5881

**Comment by myd7349** [comment]
So here is the case:

My patch is applied to `master` branch but not `v3.7.x`.

protocolbuffers/protobuf#5881 is applied to `v3.7.x` branch but not `master`.

**Comment by myd7349** [comment]
After some thoughts, I decided to switch the `HEAD_REF` to `3.7.x` in that https://github.com/protocolbuffers/protobuf/pull/5864 can be fixed temporarily with a small patch but https://github.com/protocolbuffers/protobuf/pull/5881/files needs more patching work.

By the way, I noticed that `orc` has been updated by this PR: https://github.com/Microsoft/vcpkg/pull/4786 yesterday. And it seems there are some building errors on Windows with this new version of `orc`.

**Comment by grdowns** [comment]
Hey @myd7349. I believe that `HEAD_REF` in this case should continue to point to `master`. The reason being that when users specify the `--head` option, they should always be given the latest version of regardless of its validity.

After looking over the triplet tables to see the status of `orc`, it seems that it's building fine on Windows right now. Are you able to reproduce this?

File: ports/protobuf/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: protobuf
-Version: 3.7.0-1
+Version: 2019-03-16
```
Comment (KindDragon) [inline ports/protobuf/CONTROL]: I think better name it 3.7.0-c570ece

Comment (grdowns) [reply ports/protobuf/CONTROL]: In this case, when the REF is a commit ID, we prefer to use the date at which we updated the portfile. It's tribal knowledge but we are looking to update our documentation regarding it. :grin:

**Comment by grdowns** [comment]
Hey @myd7349, I've updated the REF to point to the latest commit as opposed to the merge commit as we prefer to take the latest when possible

File: ports/protobuf/CONTROL (hunk 1 above)
Comment (KindDragon) [reply ports/protobuf/CONTROL]: It's just really difficult to understand in the output of `vcpkg upgrade` which version is `2019-03-16`.

Comment (myd7349) [reply ports/protobuf/CONTROL]: Sorry for the delay! I am on a bussiness trip these days.
@KindDragon Thanks for your review! I think it is a good idea.
@grdowns In fact, I always use the corresponding commit date of the commit. 😁

**Comment by myd7349** [comment]
Hi! @grdowns I have updated this PR.

**Comment by PhoebeHui** [comment]
Thanks for the updates! All triplets built successfully.

**Comment by myd7349** [comment]
OK. I think I should update it to 3.7.1

**Comment by myd7349** [comment]
Shall I squash these commits to keep the commit message clean?

**Pull request merged**